src/test/javascript/spec/component/overview/course-conversations/layout/conversation-header/conversation-header.component.spec.ts

```diff
@@ -164,6 +166,120 @@ examples.forEach((activeConversation) => {
+                },
+                dismiss: jest.fn(),
+                result: Promise.resolve(),
+            } as any;
+
+            const modalService = TestBed.inject(NgbModal);
+            const metisConversationService = TestBed.inject(MetisConversationService);
+            const createChatSpy = jest.spyOn(metisConversationService, 'createOneToOneChatWithId');
+
+            jest.spyOn(modalService, 'open').mockReturnValue(fakeModalRef);
+            const event = new MouseEvent('click');
+            component.openConversationDetailDialog(event, ConversationDetailTabs.INFO);
+            tick();
+
+            expect(createChatSpy).not.toHaveBeenCalled();
+        }));
+
+        it('should always open info tab when conversation is one-to-one chat', fakeAsync(() => {
+            const oneToOneChat = generateOneToOneChatDTO({});
+            component.activeConversation = oneToOneChat;
+
+            const fakeModalRef: NgbModalRef = {
+                componentInstance: {
+                    course: undefined,
+                    activeConversation: undefined,
+                    selectedTab: null,
+                    initialize: jest.fn(),
+                },
+                result: Promise.resolve(),
+                dismiss: jest.fn(),
+            } as any;
+
+            const modalService = TestBed.inject(NgbModal);
+            jest.spyOn(modalService, 'open').mockReturnValue(fakeModalRef);
+
+            const event = new MouseEvent('click');
+            component.openConversationDetailDialog(event, ConversationDetailTabs.MEMBERS);
+            tick();
+
+            expect(fakeModalRef.componentInstance.selectedTab).toEqual(ConversationDetailTabs.INFO);
+        }));
+
+        it('should emit onUpdateSidebar when conversation detail dialog is closed', fakeAsync(() => {
+            const modalService = TestBed.inject(NgbModal);
+            const fakeModalRef: NgbModalRef = {
+                componentInstance: {
+                    course: undefined,
+                    activeConversation: undefined,
+                    selectedTab: undefined,
+                    initialize: jest.fn(),
+                },
+                dismiss: jest.fn(),
+                result: Promise.resolve(),
+            } as any;
+
+            jest.spyOn(modalService, 'open').mockReturnValue(fakeModalRef);
+
+            const onUpdateSidebarSpy = jest.spyOn(component.onUpdateSidebar, 'emit');
+
+            const event = new MouseEvent('click');
+            component.openConversationDetailDialog(event, ConversationDetailTabs.INFO);
+            tick();
+
+            expect(onUpdateSidebarSpy).toHaveBeenCalled();
+        }));
+
+        it('should emit collapseSearch when toggleSearchBar is called', () => {
+            const collapseSearchSpy = jest.spyOn(component.collapseSearch, 'emit');
+            component.toggleSearchBar();
+
+            expect(collapseSearchSpy).toHaveBeenCalled();
```

Comment:
can we maybe specify the number of calls, e.g. `toHaveBeenCalledOnce()` or `toHaveBeenCalledTimes(n)`?

Reply:
I updated it to use `toHaveBeenCalledOnce()`